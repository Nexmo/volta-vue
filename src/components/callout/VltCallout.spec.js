import { shallowMount } from '@vue/test-utils';
import { VltCallout } from '..';

describe('vlt-callout', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltCallout);
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is type critical', () => {
    wrapper.setProps({ type: 'critical' });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is dismissable', () => {
    wrapper.setProps({ dismissable: true });
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is dismissed', async () => {
    wrapper.setProps({ dismissable: true });
    wrapper.find('button').trigger('click');
    expect(wrapper.emitted('dismissed')).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });
});
