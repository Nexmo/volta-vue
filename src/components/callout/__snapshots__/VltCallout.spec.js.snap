// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-callout is dismissable 1`] = `
<div
  class="Vlt-callout Vlt-callout--tip"
>
  <i />
   
  <div
    class="Vlt-callout__content"
  />
   
  <button
    class="Vlt-callout__dismiss"
  />
</div>
`;

exports[`vlt-callout is dismissed 1`] = `
<div
  class="Vlt-callout Vlt-callout--dismissed Vlt-callout--tip"
>
  <i />
   
  <div
    class="Vlt-callout__content"
  />
   
  <button
    class="Vlt-callout__dismiss"
  />
</div>
`;

exports[`vlt-callout is type critical 1`] = `
<div
  class="Vlt-callout Vlt-callout--critical"
>
  <i />
   
  <div
    class="Vlt-callout__content"
  />
   
  <!---->
</div>
`;

exports[`vlt-callout renders correctly 1`] = `
<div
  class="Vlt-callout Vlt-callout--tip"
>
  <i />
   
  <div
    class="Vlt-callout__content"
  />
   
  <!---->
</div>
`;
